main.ts

```diff
@@ -292,6 +318,84 @@ export default class FilenameHeadingSyncPlugin extends Plugin {
     return text.trim();
   }
 
+  /**
+   * Insert the `heading` at `lineNumber` in `file`.
+   *
+   * @param {TFile} file the file to modify
+   * @param {string[]} fileLines array of the file's contents, line by line
+   * @param {number} lineNumber zero-based index of the line to replace
+   * @param {string} text the new text
+   */
+  insertHeading(
+    file: TFile,
+    fileLines: string[],
+    lineNumber: number,
+    heading: string,
+  ) {
+    const newStyle = this.settings.newHeadingStyle;
+    if (newStyle === HeadingStyle.Underline) {
+      this.insertLineInFile(file, fileLines, lineNumber, `${heading}`);
+
+      this.insertLineInFile(
+        file,
+        fileLines,
+        lineNumber + 1,
+        this.settings.underlineString,
+      );
+    } else {
```

Comment:
This may be better as a switch statement, or another condition that checks for `HeadingStyle.Prefix` to make it clear that these are the available options. Prefix can still be the fallback/default